File: kirun-js/src/engine/json/schema/uiHelper/UiHelper.ts
export class UiHelper {
    private validationMessages: { [key: string]: string } = {};
    private componentPriority?: string;

    constructor(obj?: any) {
        this.validationMessages = obj?.validationMessages || {};
        this.componentPriority = obj?.componentPriority;
    }

    public setValidationMessage(type: string, message: string): UiHelper {
        this.validationMessages[type] = message;
        return this;
    }

    public getValidationMessage(type: string): string | undefined {
        return this.validationMessages[type];
    }

    public setRequiredMessage(message: string): UiHelper {
        return this.setValidationMessage('required', message);
    }

    public getRequiredMessage(): string | undefined {
        return this.getValidationMessage('required');
    }

    public setMinLengthMessage(message: string): UiHelper {
        return this.setValidationMessage('minLength', message);
    }

    public getMinLengthMessage(): string | undefined {
        return this.getValidationMessage('minLength');
    }

    public setMaxLengthMessage(message: string): UiHelper {
        return this.setValidationMessage('maxLength', message);
    }

    public getMaxLengthMessage(): string | undefined {
        return this.getValidationMessage('maxLength');
    }

    public setPatternMessage(message: string): UiHelper {
        return this.setValidationMessage('pattern', message);
    }

    public getPatternMessage(): string | undefined {
        return this.getValidationMessage('pattern');
    }

    public setFormatMessage(message: string): UiHelper {
        return this.setValidationMessage('format', message);
    }

    public getFormatMessage(): string | undefined {
        return this.getValidationMessage('format');
    }

    public setMinValueMessage(message: string): UiHelper {
        return this.setValidationMessage('minimum', message);
    }

    public getMinValueMessage(): string | undefined {
        return this.getValidationMessage('minimum');
    }

    public setMaxValueMessage(message: string): UiHelper {
        return this.setValidationMessage('maximum', message);
    }

    public getMaxValueMessage(): string | undefined {
        return this.getValidationMessage('maximum');
    }

    public setExclusiveMaxMessage(message: string): UiHelper {
        return this.setValidationMessage('exclusiveMaximum', message);
    }

    public getExclusiveMaxMessage(): string | undefined {
        return this.getValidationMessage('exclusiveMaximum');
    }

    public setExclusiveMinMessage(message: string): UiHelper {
        return this.setValidationMessage('exclusiveMinimum', message);
    }

    public getExclusiveMinimumMessage(): string | undefined {
        return this.getValidationMessage('exclusiveMinimum');
    }

    public setMultipleOfMessage(message: string): UiHelper {
        return this.setValidationMessage('multipleOf', message);
    }
    public getMultipleOfMessage(): string | undefined {
        return this.getValidationMessage('multipleOf');
    }

    public getComponentPriority(): string | undefined {
        return this.componentPriority;
    }

    public setComponentPriority(component: string): UiHelper {
        this.componentPriority = component;
        return this;
    }

    public static from(obj: any): UiHelper | undefined {
        return obj ? new UiHelper(obj) : undefined;
    }

}

File: kirun-java/src/main/java/com/fincity/nocode/kirun/engine/json/schema/uiHelper/UiHelper.java
package com.fincity.nocode.kirun.engine.json.schema.uiHelper;

import java.util.HashMap;
import java.util.Map;
import java.util.Optional;

public class UiHelper {

    private Map<String, String> validationMessages;
    private String componentPreferred;

    public UiHelper() {
        this.validationMessages = new HashMap<>();
        this.componentPreferred = null;
    }

    public UiHelper(Map<String, String> validationMessages, String componentPreferred) {
        this.validationMessages = Optional.ofNullable(validationMessages).orElse(new HashMap<>());
        this.componentPreferred = componentPreferred;
    }

    public UiHelper setValidationMessage(String type, String message) {
        this.validationMessages.put(type, message);
        return this;
    }

    public String getValidationMessage(String type) {
        return this.validationMessages.getOrDefault(type, "");
    }

    public UiHelper setRequiredMessage(String message) {
        return this.setValidationMessage("required", message);
    }

    public String getRequiredMessage() {
        return this.getValidationMessage("required");
    }

    public UiHelper setMinLengthMessage(String message) {
        return this.setValidationMessage("minLength", message);
    }

    public String getMinLengthMessage() {
        return this.getValidationMessage("minLength");
    }

    public UiHelper setMaxLengthMessage(String message) {
        return this.setValidationMessage("maxLength", message);
    }

    public String getMaxLengthMessage() {
        return this.getValidationMessage("maxLength");
    }

    public UiHelper setPatternMessage(String message) {
        return this.setValidationMessage("pattern", message);
    }

    public String getPatternMessage() {
        return this.getValidationMessage("pattern");
    }

    public UiHelper setMinValueMessage(String message) {
        return this.setValidationMessage("minimum", message);
    }

    public String getMinValueMessage() {
        return this.getValidationMessage("minimum");
    }

    public UiHelper setMaxValueMessage(String message) {
        return this.setValidationMessage("maximum", message);
    }

    public String getMaxValueMessage() {
        return this.getValidationMessage("maximum");
    }

    public UiHelper setExclusiveMinimumMessage(String message) {
        return this.setValidationMessage("exclusiveMinimum", message);
    }

    public String getExclusiveMinimumMessage() {
        return this.getValidationMessage("exclusiveMinimum");
    }

    public UiHelper setExclusiveMaximumMessage(String message) {
        return this.setValidationMessage("exclusiveMaximum", message);
    }

    public String getExclusiveMaximumMessage() {
        return this.getValidationMessage("exclusiveMaximum");
    }

    public UiHelper setMultipleOfMessage(String message) {
        return this.setValidationMessage("multipleOf", message);
    }

    public String getMultipleOfMessage() {
        return this.getValidationMessage("multipleOf");
    }

    public String getComponentPreferred() {
        return componentPreferred;
    }

    public UiHelper setComponentPreferred(String componentPreferred) {
        this.componentPreferred = componentPreferred;
        return this;
    }

    public static UiHelper from(Map<String, Object> obj) {
        if (obj == null) {
            return null;
        }

        Map<String, String> validationMessages = new HashMap<>();
        Object messagesObj = obj.get("validationMessages");

        if (messagesObj instanceof Map<?, ?>) {
            Map<?, ?> messages = (Map<?, ?>) messagesObj;
            messages.forEach((key, value) -> {
                if (key instanceof String && value instanceof String) {
                    validationMessages.put((String) key, (String) value);
                }
            });
        }

        String componentPreferred = obj.get("componentPreferred") instanceof String
                ? (String) obj.get("componentPreferred")
                : null;
        return new UiHelper(validationMessages, componentPreferred);
    }
}
